R-package/configure

```diff
@@ -1827,6 +1827,17 @@ $as_echo "${ac_pkg_openmp}" >&6; }
     fi
 fi
 
+###########
+# Solaris #
+###########
+
+# some headers used in lib_lightgbm are not available on the Solaris
+# systems CRAN tests on
+if test `uname -s` = "SunOS"
+then
+    LGB_CPPFLAGS+=" -DON_SOLARIS=1"
+fi
```

Comment:
@jameslamb 
Given that we do not need to sync this patch with CRAN submissions anymore, maybe we can simplify Solaris detection in this PR?
https://github.com/microsoft/LightGBM/pull/3338#issuecomment-698956703

Reply:
yes definitely! I'll try it here

Reply:
ok I made this change in https://github.com/microsoft/LightGBM/pull/3338/commits/c10e23dd9284ee8e703e9d7324b4bb2c947f7f29, and tested with R Hub. Worked as expected, thanks for teaching me that!